astroquery/mast/core.py

```diff
@@ -368,6 +477,488 @@ def query_object_async(self, objectname, radius=0.2*u.deg, pagesize=None, page=N
+
+        if objectname:
+            coordinates = self._resolve_object(objectname)
+
+        if coordinates:
+            # Put coordinates and radius into consitant format
+            coordinates = commons.parse_coordinates(coordinates)
+
+            # if radius is just a number we assume degrees
+            if isinstance(radius, (int, float)):
+                radius = radius * u.deg
+            radius = coord.Angle(radius)
+
+            # build the coordinates string needed by Mast.Caom.Filtered.Position
+            position = ', '.join([str(x) for x in (coordinates.ra.deg, coordinates.dec.deg, radius.deg)])
+
+        # send query
+        if position:
+            service = "Mast.Caom.Filtered.Position"
+            params = {"columns": "*",
+                      "filters": mashupFilters,
+                      "position": position}
+        else:
+            service = "Mast.Caom.Filtered"
+            params = {"columns": "*",
+                      "filters": mashupFilters}
+
+        return self.service_request_async(service, params)
+
+    def query_region_count(self, coordinates, radius=0.2*u.deg, pagesize=None, page=None):
+        """
+        Given a sky position and radius, returns the number of MAST observations in that region.
+
+        Parameters
+        ----------
+        coordinates : str or `astropy.coordinates` object
+            The target around which to search. It may be specified as a
+            string or as the appropriate `astropy.coordinates` object. 
+        radius : str or `~astropy.units.Quantity` object, optional
+            The string must be parsable by `astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from
+            `astropy.units` may also be used. Defaults to 0.2 deg.
+        pagesize : int, optional
+            Can be used to override the default pagesize for. 
+            E.g. when using a slow internet connection.
+        page : int, optional
+            Can be used to override the default behavior of all results being returned to 
+            obtain a specific page of results.
+
+        Returns
+        -------
+            response: int
+        """
+
+        # build the coordinates string needed by Mast.Caom.Filtered.Position
+        coordinates = commons.parse_coordinates(coordinates)
+
+        # if radius is just a number we assume degrees
+        if isinstance(radius, (int, float)):
+            radius = radius * u.deg
+        radius = coord.Angle(radius)
+
+        # turn coordinates into the format
+        position = ', '.join([str(x) for x in (coordinates.ra.deg, coordinates.dec.deg, radius.deg)])
+
+        service = "Mast.Caom.Filtered.Position"
+        params = {"columns": "COUNT_BIG(*)",
+                  "filters": [],
+                  "position": position}
+
+        return self.service_request(service, params, pagesize, page)[0][0].astype(int)
```

Comment:
nitpicking, but the int here types doesn't need to be numpy type, maybe instead of the ``.astype(int)``, just do ``int(...)``?